source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,343 @@
+                               Patterns::Double (),
+                               "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                              );
+            prm.declare_entry ("List of isotherms", "",
+                               Patterns::List(Patterns::List(Patterns::Anything(),0,100000000,","),0,100000000,";"),
+                               "Number of mininum and maximum areas defined by isotherms are going to be defined."
+                              );
+
+        }
+        prm.leave_subsection();
+    }
+    prm.leave_subsection();
+}
+
+template <int dim>
+void
+Isotherms<dim>::parse_parameters (ParameterHandler &prm)
+{
+    prm.enter_subsection("Mesh refinement");
+    {
+        minimum_refinement_level = Utilities::string_to_int(prm.get ("Minimum refinement level"));
+        maximum_refinement_level = Utilities::string_to_int(prm.get("Initial global refinement")) + Utilities::string_to_int(prm.get("Initial adaptive refinement"));
+        prm.enter_subsection("Isotherms");
+        {
+            exclude_composition = Utilities::string_to_int(prm.get("Exclude composition"));
+            std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+            // process the List of isotherms to get all data out of the structure
+            number_of_isotherms = isotherms_outer_loop.size();
+            isotherms.resize(number_of_isotherms,std::vector<std::string>(4,"0"));
+            // loop through all given isotherms
+            for ( unsigned int it1 = 0; it1 != number_of_isotherms; ++it1 )
+            {
+                std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[it1]);
+                // The minimum and maximum refinemnt levels respecively
+                isotherms[it1][0]=isotherms_inner_loop[0];
+                isotherms[it1][1]=isotherms_inner_loop[1];
+                // The temperatures for the min and max isotherms respectively
+                isotherms[it1][2]=isotherms_inner_loop[2];
+                isotherms[it1][3]=isotherms_inner_loop[3];
+
+            }
+        }
+        prm.leave_subsection();
+    }
+    prm.leave_subsection();
+}
+}
+}
+
+// explicit instantiations
+namespace aspect
+{
+namespace MeshRefinement
+{
+ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isotherms,
+        "isotherms",
+        "A mesh refinement criterion that ensures a "
+        "maximum and minimum refinement level between "
+        "two temperatures (isotherms), with the posibility "
+        "to exclude a composition from this criterion. "
+        "To accomplish this there are two parameters "
+        "available: 'Exclude composition' and 'List of "
+        "isotherms'. The first parameter takes the "
+        "number of the compositional field and excludes "
+        "it from the min/max refinment level prodedure. "
```

Comment:
refinment -> refinement

I don't think I understand this sentence, though. The procedure is applied to the mesh, not to a compositional field. How can you exclude a *compositional field* from the procedure?